apps/src/componentLibrary/dialog/Dialog.tsx

```diff
@@ -26,63 +39,101 @@ export interface DialogProps extends HTMLAttributes<HTMLDivElement> {
   closeLabel?: string;
   /** Dialog icon */
   icon?: FontAwesomeV6IconProps;
+  /** Dialog image url */
+  imageUrl?: string;
 }
 
 /**
  * ## Production-ready Checklist:
- *  * (?) implementation of component approved by design team;
- *  * (?) has storybook, covered with stories and documentation;
- *  * (?) has tests: test every prop, every state and every interaction that's js related;
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✔) has tests: test every prop, every state and every interaction that's js related;
  *  * (see apps/test/unit/componentLibrary/DialogTest.tsx)
  *  * (?) passes accessibility checks;
  *
- * ###  Status: ```WIP```
+ * ###  Status: ```Ready for dev```
  *
  * Design System: Dialog Component.
- * Renders Alert to notify user about something.
+ * Renders Dialog window that user should interact with.
  */
 const Dialog: React.FunctionComponent<DialogProps> = ({
   title,
-  type = 'noIcon',
-  content,
-  showSecondaryButton,
+  description,
+  primaryButtonProps,
+  secondaryButtonProps,
+  mode = 'light',
   className,
+  customContent,
+  customBottomContent,
   onClose,
   closeLabel = 'Close dialog',
   icon,
+  imageUrl,
   ...HTMLAttributes
 }) => {
+  const dialogRef = useRef<HTMLDivElement>(null);
+
+  useBodyScrollLock(true);
+  useFocusTrap(dialogRef);
+  useEscapeKeyHandler(onClose);
+
   return (
-    <div
-      className={classnames(
-        moduleStyles.dialog,
-        moduleStyles[`dialog-${type}`],
-        className
-      )}
-      {...HTMLAttributes}
-    >
-      <div>
-        {icon && <FontAwesomeV6Icon {...icon} />}
-        <span>{title}</span>
-        <span className={moduleStyles.dialogContent}>{content}</span>
-        {showSecondaryButton && (
+    <div role="presentation" className={moduleStyles.dialogOverlay}>
+      <div
```

Comment:
Is there a reason you opted to use a div here instead of the `<dialog>` element? ([Mozilla ref](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog))

Reply:
Good question. Mostly because I'm not sure if all the browsers we support supports `<dialog>`, plus trying to avoid throwing refs back and forth between components instead of handling all the logic from parent component

Reply:
That makes sense!

Reply:
Double checked this question, here's what I've found out:
1. (+) Turned out it should be relatively easy to convert `Dialog` and `Modal` to use <dialog>. Here's a little snippet to get the main idea of work to be done for this:
```
...
const Dialog: React.FunctionComponent<DialogProps> = ({
  title,
  description,
  primaryButtonProps,
  secondaryButtonProps,
  mode = 'light',
  className,
  customContent,
  customBottomContent,
  onClose,
  closeLabel = 'Close dialog',
  icon,
  imageUrl,
  ...HTMLAttributes
}) => {
  const dialogRef = useRef<HTMLDialogElement>(null);

  // Open the dialog on mount
  useEffect(() => {
    const dialog = dialogRef.current;
    if (dialog) dialog.showModal();
  }, []);

  // Handle the close action
  const handleClose = () => {
    dialogRef.current?.close();
    if (onClose) onClose();
  };

  useEscapeKeyHandler(handleClose);

  return (
    <dialog
      ref={dialogRef}
      aria-labelledby="dsco-dialog-title"
      aria-describedby="dsco-dialog-description"
      className={classnames(
        moduleStyles.dialog,
        moduleStyles[`dialog-${mode}`],
        className
      )}
      {...HTMLAttributes}
    >
      <div className={moduleStyles.dialogTextSection}>
        {imageUrl && <img src={imageUrl} alt="Dialog" />}
        <Heading2 id="dsco-dialog-title">{title}</Heading2>
        {description && (
        ...
```
This will also require updating styles, but that's not that big of a deal cause it should be mostly changing css selectors.


2. (-) I've checked the list of [Code.org supported browser](https://code.org/educate/it) and there's a problem with using `<dialog>` - Code.org supports Safari `14.x` minimum and Safari mobile `14.x` minimum, meanwhile [<dialog> is supported in both Safari and Safari mobile starting with](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#browser_compatibility) `15.4`

Reply:
Giving this, I suggest we use <div> for now and once we'll support 15.4+ Safari and Safari mobile - revisit this question

Reply:
Thanks for looking into this further! I think that's a fair assessment if we still support Safari 14.